content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+
+## Main routine
+
+As a result, the Python main routine of the ILO Prometheus exporter is as simple as (I removed some of the log commands to improve the readability):
+
+```python
+if __name__ == "__main__":
+    input = getServerList()    
+    # Get the monitoring URLs of the server
+    monitor_urls = get_server_urls(input['user'], input['password'], input['server'], input['lfile'])
+    # Start the http_server and the counters, gauges
+    start_http_server(input['port'])
+    c = Counter('ilorest_sample','ILO REST sample number')
+    node = Gauge('ilorest_node','ILO Node Data',['nodename','rack','nodemetric','metricdetail'])
+    delta = Gauge('ConnectorRuntime','Time required for last data collection in seconds')
+    inode = Info('node','Additional Node Info',['node'])
+    # Start the endless loop
+    while True: 
+        t0 = time.time()
+        start0 = t0
+        c.inc()      
+        for server in monitor_urls:
+            try:
+                server_metrics = get_server_data(input['user'], input['password'], monitor_urls[server], input['lfile'])
+                display_results(node, inode, server_metrics, monitor_urls[server])
+            except Exception as ex:
+                log=logopen(input['lfile'])
+                logwriter(log,'Exception')
+                logwriter(log,str(ex.__context__))
+                logclose(log)
+                pass
+        t1 = time.time()
+        delta.set((t1-t0))
+        while ((t1-t0) < input['mintervall']):
+            time.sleep(1.0)
+            t1 = time.time()   
+        # update once per day the input
+        if ( t1 - start0 > 86400):
+            start0 = t1
+            input = getServerList() 
+            monitor_urls = get_server_urls(input['user'], input['password'], input['server'], input['lfile'])
+```
+
+As you can see, I used four Python functions within the main routine:
+
+* *getServerList*
+* *get\_server\_urls*
+* *get\_server\_data*
+* *display\_results*
+
+The first one - *getServerList* - is just reading the input data out of an XML file. Since this is a pretty standard task in Python I do not want to discuss this function here further. It should be sufficient to know, that it is reading the list of server IPs and additional runtime parameters (username, password, monitoring intervall, …) into a Python directory, that is used in the subsequent parts of the script. I want to focus on the other three functions, that are more specific to the ILO Prometheus exporter.
```

Comment:
Place a comma after "standard task in Python".
Remove the comma between "to know" and "that is reading".
Also remove the comma between Python directory" and "that is used in"
And remove the comma between "three functions" and "that are more"